internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -655,6 +655,27 @@ func getHyperthreading(clusterInstall *hiveext.AgentClusterInstall) *string {
 	}
 }
 
+func (r *ClusterDeploymentsReconciler) getEncodedCACert(ctx context.Context,
+	log logrus.FieldLogger,
+	clusterDeployment *hivev1.ClusterDeployment,
+	clusterInstall *hiveext.AgentClusterInstall) (*string, error) {
+	caCertificateRef := clusterInstall.Spec.IgnitionEndpoint.CaCertificateReference
+	if caCertificateRef != nil {
+		caSecret, err := getSecret(ctx, r.Client, r.APIReader, types.NamespacedName{Namespace: caCertificateRef.Namespace, Name: clusterDeployment.Spec.PullSecretRef.Name})
+		if err != nil {
+			return swag.String(""), errors.Wrap(err, "error getting ca certificate secret")
+		}
+		caCertBytes, hasCACert := caSecret.Data[corev1.TLSCertKey]
```

Comment:
Need documentation on the Secret data

Reply:
It means you should document how this option should be used (a working example is best).
e.g. https://github.com/openshift/assisted-service/blob/5e759094ebf7e6a4f95f30c9404b964754079cad/docs/hive-integration/README.md#creating-host-installer-args-overrides

Reply:
Still missing documentation...

Reply:
I added agentClusterInstall-with-ignitionEndpoint.yaml

Reply:
That is great, but how will a user know what should be the content of the caCertificate secret (Or that the caCertificate should be a secret to begin with)?
```
  caCertificateReference:
    namespace: "spoke-cluster"
    name: "secret-name"
```

Reply:
Done